File: ASP.NET Core + Angular/Task/OnlineBookStoreAPI/Helpers/BookParams.cs
using OnlineBookStoreAPI.Constants;

namespace OnlineBookStoreAPI.Helpers
{
    public class BookParams : PaginationParams
    {
        public int MinPrice { get; set; } = 0;
        public int MaxPrice { get; set; } 
        public string SortBy { get; set; } = Const.TITLE;
        public string SortOrder { get; set; } = Const.ASCENDING;

    }
}

File: ASP.NET Core + Angular/Task/client/src/app/_models/bookParams.ts
import { Constants } from "./constants";

export class BookParams {
    minPrice: number;
    maxPrice: number;
    pageNumber: number;
    pageSize: number;
    sortBy: string;
    sortOrder: string;

    constructor(sortBy?: string, sortOrder?:string, minPrice?: number, maxPrice?:number, pageNumber?: number, pageSize?:number) {
        this.minPrice = minPrice || Constants.minPrice;
        this.maxPrice = maxPrice || Constants.maxPrice;
        this.pageNumber = pageNumber || Constants.pageNumber;
        this.pageSize = pageSize || Constants.pageSizeOptions[4];
        this.sortBy = sortBy || Constants.sortByTitle;
        this.sortOrder = sortOrder || Constants.sortOrderAsc;
    }
}

